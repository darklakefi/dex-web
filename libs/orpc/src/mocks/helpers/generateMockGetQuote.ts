import {
  randBoolean,
  randCurrencySymbol,
  randNumber,
  randUuid,
  seed,
} from "@ngneat/falso";
import type {
  GetQuoteInput,
  GetQuoteOutput,
} from "../../schemas/swaps/getQuote.schema";

export function generateMockGetQuote(input: GetQuoteInput) {
  seed(
    `${input.amountIn.toString()}${input.isXtoY.toString()}${input.slippage.toString()}${input.tokenX}${input.tokenY}${input.poolAddress}`,
  );

  return {
    amountOut: randNumber({ max: 1000000, min: 1 }),
    deadline: randNumber({ max: 1000000, min: 1 }),
    estimatedFee: randNumber({ max: 1000000, min: 1 }),
    estimatedFeesUsd: randNumber({ max: 1000000, min: 1 }),
    isXtoY: randBoolean(),
    poolAddress: randUuid(),
    priceImpactPercentage: randNumber({ max: 100, min: 1 }),
    rateXtoY: randNumber({ max: 1000000, min: 1 }),
    slippage: randNumber({ max: 100, min: 1 }),
    tokenX: {
      address: randUuid(),
      symbol: randCurrencySymbol(),
      value: randNumber({ max: 1000000, min: 1 }).toString(),
    },
    tokenY: {
      address: randUuid(),
      symbol: randCurrencySymbol(),
      value: randNumber({ max: 1000000, min: 1 }).toString(),
    },
    userAddress: randUuid(),
  } satisfies GetQuoteOutput;
}
