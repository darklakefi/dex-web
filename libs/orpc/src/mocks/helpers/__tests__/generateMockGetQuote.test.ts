import { generateMockGetQuote } from "../generateMockGetQuote";

describe("generateMockGetQuote", () => {
  it("should generate a quote", () => {
    const response = generateMockGetQuote({
      amountIn: 100,
      isXtoY: true,
      poolAddress: "poolAddress",
      slippage: 1,
      tokenX: "tokenX",
      tokenY: "tokenY",
    });

    expect(response).toMatchSnapshot();
    expect(response.tokenX).toBeDefined();
    expect(response.tokenY).toBeDefined();
    expect(response.poolAddress).toBeDefined();
    expect(response.amountOut).toBeDefined();
    expect(response.isXtoY).toBeDefined();
    expect(response.rateXtoY).toBeDefined();
    expect(response.slippage).toBeDefined();
    expect(response.estimatedFee).toBeDefined();
    expect(response.estimatedFeesUsd).toBeDefined();
    expect(response.priceImpactPercentage).toBeDefined();
    expect(response.deadline).toBeDefined();
    expect(response.userAddress).toBeDefined();
  });
});
