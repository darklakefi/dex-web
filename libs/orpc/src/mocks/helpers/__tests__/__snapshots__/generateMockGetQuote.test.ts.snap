// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateMockGetQuote > should generate a quote 1`] = `
{
  "amountOut": 921062,
  "deadline": 655453,
  "estimatedFee": 560177,
  "estimatedFeesUsd": 325667,
  "isXtoY": false,
  "poolAddress": "ccd7af3a-e849-4d09-9540-b27b437b5e60",
  "priceImpactPercentage": 81,
  "rateXtoY": 1324,
  "slippage": 34,
  "tokenX": {
    "address": "e78cf7a8-fe06-4d1d-9474-870c7f047dcc",
    "symbol": "â‚¡",
    "value": "316153",
  },
  "tokenY": {
    "address": "3e84bbe0-16cf-4629-810d-129a769ed8eb",
    "symbol": "p.",
    "value": "656865",
  },
  "userAddress": "1021e76f-dc65-4462-90bf-6098424d0676",
}
`;
